src/shared/lib/StoreUtils.ts

```diff
@@ -846,14 +846,20 @@ export async function fetchStructuralVariantOncoKbData(
     ) {
         return ONCOKB_DEFAULT;
     } else {
-        const alterationsToQuery = _.filter(
-            structuralVariantData.result,
-            d =>
+        const alterationsToQuery = _.filter(structuralVariantData.result, d => {
+            // We assume when one of the gene partners is missing, it should be intragenic deletion.
+            const isIntragenic =
+                d.variantClass &&
+                (d.site1EntrezGeneId === d.site2EntrezGeneId ||
+                    d.site1EntrezGeneId === undefined ||
+                    d.site2EntrezGeneId === undefined);
+            const otherValidQuery =
```

Comment:
this name could be improved.  like, what about, ```isAnnotated``` or ```atLeastOneAnnotated```

i also think using the ```in``` operator would be clearer than ```!!```

And this validation filter should really be have a unit test.

Reply:
All valid points. Before making any changes, I do want to get some feedback from @kalletlak. I do believe intragenic events should be properly marked with Deletion as VariantClass. But it looks like it's not always the case. So I do not check the variant class anymore. As long as one of the genes is missing, I consider it intragenic.